extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -23,6 +23,9 @@
 import io.agroal.api.security.SimplePassword;
 import io.agroal.api.transaction.TransactionIntegration;
 import io.agroal.narayana.NarayanaTransactionIntegration;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.vault.CredentialProvider;
```

Comment:
Same here, I think it should be `credentials`. @sberyozkin @sebastienblanc you're more in the security domain than I am, can you check?

Here I think we are only talking about the password, not the username/password so maybe singular is OK? I'm not sure about it myself but as it will be a widely used API, we need to get this right.

Reply:
initially it was credential-provider. and I renamed it `credential` to be more compact in the config. it is not the credentials directly, but a producer of credentials.
so for instance DataSourceRuntimeConfig.credential is the name of the credential provider to use.
I can rename it `credential-provider` if you think that is more appropriate.
in the meantime I will go with your suggestion of renaming it `credentials`

Reply:
initially it was `credential-provider`. and I renamed it `credential` to be more compact in the config. it is not the credentials directly, but a producer of credentials.
so for instance DataSourceRuntimeConfig.credential is the name of the credential provider to use.
I renamed it `credential-provider` to make it clearer.